packages/commonwealth/client/scripts/views/pages/Community/Members/CommunityMembersPage.tsx

```diff
@@ -0,0 +1,218 @@
+    500
+  );
+
+  const { data: members, fetchNextPage } = useSearchProfilesQuery({
+    chainId: app.activeChainId(),
+    searchTerm: '',
+    limit: 10,
+    orderBy: APIOrderBy.LastActive,
+    orderDirection: APIOrderDirection.Desc,
+    includeRoles: true,
+  });
+
+  const { data: groups } = useFetchGroupsQuery({
+    chainId: app.activeChainId(),
+    shouldIncludeMembers: true,
+  });
+
+  const formattedMembers = useMemo(() => {
+    if (!members?.pages?.length) {
+      return [];
+    }
+
+    return members.pages
+      .reduce((acc, page) => {
+        return [...acc, ...page.results];
+      }, [] as SearchProfilesResponse['results'])
+      .map((p) => ({
+        name: p.profile_name || 'Anonymous',
+        role: p.roles.find(
+          (role) =>
+            role.chain_id === app.activeChainId() &&
+            [Permissions.ROLES.ADMIN, Permissions.ROLES.MODERATOR].includes(
+              role.permission
+            )
+        )?.permission,
+        groups: (groups || [])
+          .filter((g) =>
+            (g.members || []).find(
+              (x) => x?.address?.address === p.addresses?.[0]?.address
+            )
+          )
+          .map((x) => x.name),
+      }))
+      .filter((p) =>
+        debouncedSearchTerm
+          ? p.groups.find((g) =>
+              g.toLowerCase().includes(debouncedSearchTerm.toLowerCase())
+            ) ||
+            p.name.toLowerCase().includes(debouncedSearchTerm.toLowerCase())
+          : true
+      );
+  }, [members, groups, debouncedSearchTerm]);
+
+  const totalResults = members?.pages?.[0]?.totalResults || 0;
+
+  // fixes bug that prevents scrolling on initial page load
+  useEffect(() => {
+    const shouldFetchMore = formattedMembers.length < 50 && totalResults > 50;
+    if (!shouldFetchMore) {
+      return;
+    }
+    fetchNextPage();
+  }, [formattedMembers, totalResults, fetchNextPage]);
+
+  useEffect(() => {
+    // Set the active tab based on URL
+    if (
+      window.location.search.includes(
+        `tab=${TABS[0].split(' ').join('-').toLowerCase()}`
+      )
+    ) {
+      setSelectedTab(TABS[0]);
+    } else if (
+      isGatingEnabled &&
+      window.location.search.includes(
+        `tab=${TABS[1].split(' ').join('-').toLowerCase()}`
+      )
+    ) {
+      setSelectedTab(TABS[1]);
+    } else {
+      setSelectedTab(TABS[0]);
+    }
+  }, []);
+
+  const navigateToCreateGroupPage = () => {
+    navigate(`/${app.activeChainId()}/members/groups/create`);
+  };
+
+  const isAdmin = Permissions.isCommunityAdmin() || Permissions.isSiteAdmin();
+
+  return (
+    <section className="CommunityMembersPage">
+      {/* TODO: add breadcrums here */}
+
+      {/* Header */}
+      <CWText type="h2">Members ({totalResults})</CWText>
+
+      {/* Tabs section */}
+      <CWTabBar>
+        {TABS.map((tab, index) => (
+          <CWTab
+            key={index}
+            label={tab}
+            onClick={() => setSelectedTab(tab)}
```

Comment:
there is a logic for setting active tab based on URL but on tab click query param is not added to the url 🤔 


![image](https://github.com/hicommonwealth/commonwealth/assets/14819225/58b0aa89-ae8a-41ad-a25e-8f71374e6ed3)

Reply:
i did add that code in some pr, but I think I removed it somewhere. Added again